### PR TITLE
Jersey upgrade

### PR DESCRIPTION
**Pull request by raipc**
Reasons to this refactoring: multiple warnings from snyk.io, uncomfortable way to access config singleton

### CODE REVIEW

File: src/main/java/com/axibase/tsd/api/method/MethodParameters.java [hunk 1]
```diff
@@ -43,19 +41,15 @@ private static String formatParameterName(Iterable<String> parameterName) {
         return String.join(".", parameterName);
     }
 
+    @SuppressWarnings("unchecked")
     private static String formatCollection(Collection collection) {
```
Comment (unrealwork) [inline src/main/java/com/axibase/tsd/api/method/MethodParameters.java]: Maybe the variant is more readable

```java
private static String formatCollection(Collection<?> collection) {
        return collection.stream()
                .filter(Objects::nonNull)
                .map(Object::toString)
                .collect(Collectors.joining(","));

    }
```

File: src/main/java/com/axibase/tsd/api/method/csv/CSVUploadMethod.java [hunk 2]
```diff
@@ -26,38 +27,45 @@ public static File resolvePath(String path) throws URISyntaxException, FileNotFo
...
-        MultiPart multiPart = new MultiPart();
-        MediaType mediaType = getMediaTypeFromFile(file);
-        if (mediaType.equals(MediaType.APPLICATION_OCTET_STREAM_TYPE)) {
-            mediaType = new MediaType("text", "csv");
-        }
-        FileDataBodyPart fileDataBodyPart = new FileDataBodyPart("filedata", file, mediaType);
-        multiPart.bodyPart(fileDataBodyPart);
+        try (MultiPart multiPart = new MultiPart()) {
+            MediaType mediaType = getMediaTypeFromFile(file);
+            if (mediaType.equals(MediaType.APPLICATION_OCTET_STREAM_TYPE)) {
```
Comment (unrealwork) [inline src/main/java/com/axibase/tsd/api/method/csv/CSVUploadMethod.java]: Flip the `equals` for null-safety.

File: src/main/java/com/axibase/tsd/api/transport/tcp/TCPSender.java [hunk 3]
```diff
@@ -20,41 +20,27 @@
...
              DataOutputStream requestStream = new DataOutputStream(socket.getOutputStream());
              BufferedReader responseStream = new BufferedReader(
                      new InputStreamReader(socket.getInputStream(), StandardCharsets.UTF_8)
              )) {
-            String request = isDebugMode ? DEBUG_PREFIX.concat(command) : command;
+
             IOUtils.write(request, requestStream);
             String response = isDebugMode ? responseStream.readLine() : null;
-            LOGGER.debug(" > tcp://{}:{} \n\t{}\n< {}", config.getServerName(), config.getTcpPort(), request, response);
+            LOGGER.debug(" > tcp://{}:{} \n\t{}\n< {}", host, port, request, response);
```
Comment (unrealwork) [inline src/main/java/com/axibase/tsd/api/transport/tcp/TCPSender.java]: We should choose the name convention for loggers. Better migrate to Lombok as in other classes.

File: src/test/java/com/axibase/tsd/api/util/TestUtil.java [hunk 4]
```diff
@@ -248,6 +248,10 @@ public static String quoteEscape(String s) {
         return Arrays.stream(data).map(o -> new Object[] {o}).toArray(Object[][]::new);
     }
 
+    public static Object[][] convertTo2DimArray(Collection<?> data) {
+        return data.stream().map(o -> new Object[] {o}).toArray(Object[][]::new);
```
Comment (unrealwork) [inline src/test/java/com/axibase/tsd/api/util/TestUtil.java]: You can use `ArrayUtils::toArray` from Apache Commons instead raw array creation

File: src/main/java/com/axibase/tsd/api/transport/tcp/TCPSender.java (hunk 3 above)
Comment (raipc) [reply src/main/java/com/axibase/tsd/api/transport/tcp/TCPSender.java]: I didn't modify the logger name. It should be done in the other PR

**Pull request merged**